### PR TITLE
LPS-76201 FAKE GRADLE CACHE

### PR DESCRIPTION
**Pull request by gamerson**

### CODE REVIEW

**Comment by gamerson** [comment]
ci:reopen

File: modules/sdk/gradle-plugins-soy/build.gradle [hunk 1]
```diff
@@ -24,4 +24,4 @@ gradleTest {
 	dependsOn jar
 
 	versions "2.5", "2.6", "2.7", "2.8", "2.9", "2.10", "2.11", "2.12", "2.13", "2.14.1", "3.0", "3.1", "3.2.1", "3.3"
-}
\ No newline at end of file
+}
```
Comment (Ithildir) [inline modules/sdk/gradle-plugins-soy/build.gradle]: SF problem.

File: modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/archetype-resources/pom.xml [hunk 2]
```diff
@@ -69,88 +69,55 @@
...
-								<argument>--soySrc</argument>
-								<argument>**/*.soy</argument>
-								<argument>--src</argument>
-								<argument>**/*.es.js</argument>
-								<argument>**/*.soy.js</argument>
-							</arguments>
+							<nodePath>${project.build.directory}/node/node</nodePath>
+							<workingPath>${project.build.outputDirectory}/META-INF/resources</workingPath>
+							<modulePath>${node.modules.dir}/metal-cli/index.js</modulePath>
+							<dest>${project.build.outputDirectory}/META-INF/resources</dest>
```
Comment (Ithildir) [inline modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/archetype-resources/pom.xml]: Can't we sort these?

File: modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/archetype-resources/pom.xml [hunk 3]
```diff
@@ -69,88 +69,55 @@
...
-								<argument>--namespace</argument>
-								<argument>Liferay.Loader</argument>
-								<argument>--output</argument>
-								<argument>${project.build.outputDirectory}/META-INF/config.json</argument>
-								<argument>--moduleRoot</argument>
-								<argument>${project.build.outputDirectory}/META-INF/resources</argument>
-								<argument>${project.build.outputDirectory}/META-INF</argument>
-							</arguments>
+							<nodePath>${project.build.directory}/node/node</nodePath>
+							<workingPath>${project.build.outputDirectory}/META-INF/resources</workingPath>
```
Comment (Ithildir) [inline modules/sdk/project-templates/project-templates-soy-portlet/src/main/resources/archetype-resources/pom.xml]: Can't we sort these?

File: modules/util/portal-tools-soy-builder/src/main/java/com/liferay/portal/tools/soy/builder/commands/BaseNodeCommand.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	public void setWorkingPath(String workingPath) {
+		_workingPath = workingPath;
+	}
+
+	protected static int executeCommand(String command, String workingPath)
+		throws InterruptedException, IOException {
+
+		File file = new File(workingPath);
+
+		final Runtime runtime = Runtime.getRuntime();
```
Comment (Ithildir) [inline modules/util/portal-tools-soy-builder/src/main/java/com/liferay/portal/tools/soy/builder/commands/BaseNodeCommand.java]: Please don't use `final` if you don't need it.

File: modules/util/portal-tools-soy-builder/src/main/java/com/liferay/portal/tools/soy/builder/commands/BaseNodeCommand.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+		description = "The path to the node executable.",
+		names = {"--nodePath"}, required = true
+	)
+	private String _nodePath;
+
+	@Parameter(
+		description = "The working path (directory) of the process to be executed.",
+		names = {"--workingPath"}, required = true
+	)
+	private String _workingPath;
```
Comment (Ithildir) [inline modules/util/portal-tools-soy-builder/src/main/java/com/liferay/portal/tools/soy/builder/commands/BaseNodeCommand.java]: Can't we use `File` here?

File: modules/util/portal-tools-soy-builder/src/main/java/com/liferay/portal/tools/soy/builder/commands/BaseNodeCommand.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+		description = "The path to the node module to be executed.",
+		names = {"--modulePath"}, required = true
+	)
+	private String _modulePath;
+
+	@Parameter(
+		description = "The path to the node executable.",
+		names = {"--nodePath"}, required = true
+	)
+	private String _nodePath;
```
Comment (Ithildir) [inline modules/util/portal-tools-soy-builder/src/main/java/com/liferay/portal/tools/soy/builder/commands/BaseNodeCommand.java]: Can't we use `File` here?

File: modules/util/portal-tools-soy-builder/src/main/java/com/liferay/portal/tools/soy/builder/commands/BaseNodeCommand.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+		final int returnCode = process.waitFor();
+
+		return returnCode;
+	}
+
+	@Parameter(
+		description = "The path to the node module to be executed.",
+		names = {"--modulePath"}, required = true
+	)
+	private String _modulePath;
```
Comment (Ithildir) [inline modules/util/portal-tools-soy-builder/src/main/java/com/liferay/portal/tools/soy/builder/commands/BaseNodeCommand.java]: Can't we use `File` here?

File: modules/util/portal-tools-soy-builder/src/main/java/com/liferay/portal/tools/soy/builder/commands/TranspileJsCommand.java [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+/**
+ * @author Christopher Bryan Boyd
+ */
+@Parameters(
+	commandDescription = "Invoke metal CLI to transpile Soy to JS.",
+	commandNames = "transpile-js"
+)
+public class TranspileJsCommand extends BaseNodeCommand {
+
+	public TranspileJsCommand() {
```
Comment (Ithildir) [inline modules/util/portal-tools-soy-builder/src/main/java/com/liferay/portal/tools/soy/builder/commands/TranspileJsCommand.java]: Do we need this constructor?

**Review by Ithildir — CHANGES_REQUESTED** [review]

**Comment by Ithildir** [comment]
Hi @christopherbryanboyd and @gamerson, can you please address these problems and resend? Thanks!

**Comment by gamerson** [comment]
will do

**Comment by christopherbryanboyd** [comment]
Hi @Ithildir , I made these changes you requested and sent you an updated PR here: https://github.com/Ithildir/liferay-portal/pull/845/

**Pull request closed without merging**